packages/adminrouter/extra/src/nginx.master.conf

```diff
@@ -296,7 +296,25 @@ http {
             }
             include common/proxy-headers.conf;
             proxy_pass http://mesos_dns/;
+        }
+
+        location /internal/mesos_dns/ {
+            # This location is meant to be used internally in order to resolve
+            # MesosDNS SRV records:
+            # http://nginx.org/en/docs/http/ngx_http_core_module.html#internal
+            internal;
+
+            # Do not send original request headers upstream, see
+            # https://github.com/openresty/lua-nginx-module#ngxlocationcapture
+            proxy_pass_request_headers off;
 
+            include common/proxy-headers.conf;
+
+            # On average case, maximum time user is willing to wait for
+            # response is around 5s. Basing on that these timeouts were chosen.
```

Comment:
@adamtheturtle wants it here, you do not like it, and I am just sitting here repushing and repushing and repushing the integration tests.

I really do not mind fixing it/removing it - but for the sake of saving some of my time, let's do it in the next PR and not destroy the hard-won all-green status of this PR :D